sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2023,4 +2023,11 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       assert(df1.join(df2, $"t1.i" === $"t2.i").cache().count() == 1)
     }
   }
+
+  test("order-by ordinal.") {
+    val df = Seq((1, 1), (1, 2), (2, 1), (2, 2), (3, 1), (3, 2)).toDF("a", "b")
+    checkAnswer(
+      df.select(lit(7), 'a, 'b).orderBy(lit(1), lit(2), lit(3)),
+      Seq(Row(7, 1, 1), Row(7, 1, 2), Row(7, 2, 1), Row(7, 2, 2), Row(7, 3, 1), Row(7, 3, 2)))
```

Comment:
OK. I see. When we input query like `df.select(lit(7), 'a, 'b).orderBy(lit(1), lit(2), lit(3))`, the query plan looks like:

    Sort [7#22 ASC NULLS FIRST, a#5 ASC NULLS FIRST, b#6 ASC NULLS FIRST], true
    +- Project [7 AS 7#22, a#5, b#6]
       +- Project [_1#2 AS a#5, _2#3 AS b#6]
          +- LocalRelation [_1#2, _2#3]

We have a `Project` below `Sort`. The ordinal `1` be replaced with the attribute `7#22`. So we won't get an int literal 7 here. That is why it passes.

Can you have a test for ordinal order-by that show different behavior? If not, I think ordinal order-by should be safe from this bug. And we don't need to add this test.